src/tty/unix.rs

```diff
@@ -277,12 +253,14 @@ pub fn new<T: ToWinsize>(
         }
 
         unsafe {
-            libc::signal(libc::SIGCHLD, libc::SIG_DFL);
-            libc::signal(libc::SIGHUP, libc::SIG_DFL);
-            libc::signal(libc::SIGINT, libc::SIG_DFL);
-            libc::signal(libc::SIGQUIT, libc::SIG_DFL);
-            libc::signal(libc::SIGTERM, libc::SIG_DFL);
-            libc::signal(libc::SIGALRM, libc::SIG_DFL);
+            let dfl = SigAction::new(SigHandler::SigDfl, SaFlags::empty(), SigSet::empty());
+            sigaction(Signal::SIGCHLD, &dfl).unwrap();
+            sigaction(Signal::SIGHUP,  &dfl).unwrap();
+            sigaction(Signal::SIGINT,  &dfl).unwrap();
+            sigaction(Signal::SIGQUIT, &dfl).unwrap();
+            sigaction(Signal::SIGTERM, &dfl).unwrap();
+            sigaction(Signal::SIGALRM, &dfl).unwrap();
+            sigaction(Signal::SIGPIPE, &dfl).unwrap();
```

Comment:
Just for documentation, the `sigaction` call will fail when either the signal is invalid or a signal that can't be caught, or the `SaFlags` contain `SA_SIGINFO`. Since the flags are empty and the signals are constant, it is safe to unwrap here and guaranteed to work (assuming it was tested at least once and none of the signals were illegal).